examples/centralized_threshold.rs

```diff
@@ -0,0 +1,172 @@
+
+        // Create the Alba proof using the prover set
+        let alba = CentralizedTelescope::create(params);
+        let proof = alba.prove(&prover_set)?;
+
+        // Extract the registration indices of the elements that form the proof element sequence
+        let indices: Vec<usize> = proof
+            .element_sequence
+            .iter()
+            .filter_map(|element| valid_signatures.get(element.as_slice()).copied())
+            .collect();
+
+        // Return the constructed AlbaThresholdSignature
+        Some(Self {
+            proof,
+            indices,
+            commitment,
+        })
+    }
+
+    /// Verify AlbaThresholdSignature. Validate individual signatures and verify Alba proof.
+    fn verify<const N: usize>(
+        &self,
+        params: &Params,
+        registration: &Registration,
+        msg: &[u8],
+    ) -> bool {
+        // Check if the checksum exists in the registration
+        let Some(checksum) = &registration.checksum else {
+            println!("Error: Registration is not closed.");
+            return false;
+        };
+
+        // Compute the commitment and compare with the stored commitment
+        let commitment = get_commitment::<N>(checksum, msg).to_vec();
+        if commitment != self.commitment {
+            println!("Error: Commitment mismatch.");
+            return false;
+        }
+
+        // Validate the individual signatures
+        if !validate_signatures(self, registration, &commitment) {
+            println!("Error: Signature validation failed.");
+            return false;
+        }
+
+        // Create the Alba proof and verify it
+        let alba = CentralizedTelescope::create(params);
+        let result = alba.verify(&self.proof);
+
+        if result {
+            println!("Success: Alba proof verification passed.");
+        } else {
+            println!("Error: Alba proof verification failed.");
+        }
+
+        result
+    }
+}
+
+fn main() {
+    let mut rng = ChaCha20Rng::from_seed(Default::default());
+    let mut msg = [0u8; 16];
+    rng.fill_bytes(&mut msg);
+    let set_size = 1_000;
+    let params = Params {
+        soundness_param: 10.0,
+        completeness_param: 10.0,
+        set_size: 80 * set_size / 100,
+        lower_bound: 20 * set_size / 100,
+    };
+
+    // Initialize signers
+    let signers: Vec<Signer> = (0..set_size as usize)
+        .map(|_| Signer::init(&mut rng))
+        .collect();
+
+    // Start new registration process
+    let mut registration = Registration::new();
+
+    // Register signers
+    for (index, signer) in signers.iter().enumerate() {
+        registration.register(signer.verification_key, index);
+    }
+    // Close registration to compute the checksum
+    registration.close::<DATA_LENGTH>();
+
+    // Filter out registered signers and create RegisteredSigner instances
+    let registered_signers: Vec<RegisteredSigner> = signers
```

Comment:
Why do you filter here the signers? Either they are all registered (honest scenario), or they are not (dishonest scenario) but we still want to check that the aggregator doesn't use these signatures.